homeassistant/config_entries.py

```diff
@@ -2399,15 +2399,14 @@ def async_update_reload_and_abort(
         reason: str = "reauth_successful",
     ) -> ConfigFlowResult:
         """Update config entry, reload config entry and finish config flow."""
-        result = self.hass.config_entries.async_update_entry(
+        self.hass.config_entries.async_update_entry(
             entry=entry,
             unique_id=unique_id,
             title=title,
             data=data,
             options=options,
         )
-        if result:
-            self.hass.config_entries.async_schedule_reload(entry.entry_id)
+        self.hass.config_entries.async_schedule_reload(entry.entry_id)
```

Comment:
This makes sense as the data could be the same, but the state of the device/service could be different (which we won't know) and we only call this if there is new data (well at least in most cases)